### ISSUE TITLE
RSpec is failuring on Float comparison

### ISSUE THREAD

**Issue by abinoam** [mrhead/ruby_arena#32]
Failures:
  1) Robot#tick decreases gun heat by 0.1 if it is more than 0
     Failure/Error: expect(robot.gun_heat).to be 0.9

```
   expected #<Float:70150894016120> => 0.9
        got #<Float:70150893189340> => 0.9

   Compared using equal?, which compares object identity,
   but expected and actual are not the same object. Use
   `expect(actual).to eq(expected)` if you don't care about
   object identity in this example.
 # ./spec/robot_spec.rb:22:in `block (3 levels) in <top (required)>'
```

**Comment by abinoam** [comment]
Working on it.

**Comment by mrhead** [comment]
Hmm, I have everything green, so maybe it is because of different RSpec version.

**Comment by mrhead** [comment]
I've merged your pull request, so I presume that this is solved.

Anyway there is now this warning for RSpec 3.0.0.beta1:

```
Deprecation Warnings:

Using `any_instance` from rspec-mocks' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` instead. Called from /Users/head/Dropbox/rails/ruby_arena/spec/level_spec.rb:5:in `block (2 levels) in <top (required)>'.


If you need more of the backtrace for any of these deprecations to
identify where to make the necessary changes, you can configure
`config.raise_errors_for_deprecations!`, and it will turn the
deprecation warnings into errors, giving you the full backtrace.
```

Hmm, do you know how to fix this for the new RSpec. 

Also I probably should specify exact RSpec version in Gemfile...

**Comment by abinoam** [comment]
Float comparison is tricky (from long ago).

1.0 - 0.1 renders a representation of a float that is very close to 0.9 but it's not guaranteed to be equal to 0.8 + 0.1 for example, or 0.9 itself.

I'm not sure, but perhaps there's another "oficial" way to compare floats.
Something like
be_close
be_within
http://rubydoc.info/gems/rspec-expectations/2.4.0/RSpec/Matchers

But, this commit made my tests green anyway.
I just did as suggested by the error message of RSpec.

**Comment by orendon** [comment]
Just fixed the warning on PR #34 specifically this commit https://github.com/orendon/ruby_arena/commit/40b32d78ad3e5c03d424e283dd6034d20e8481cf

**Comment by mrhead** [comment]
Thanks to both!
